src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -529,85 +527,52 @@ public virtual void WriteLiteralTo([NotNull] TextWriter writer, string value)
             }
             else
             {
-                for (var i = 0; i < values.Length; i++)
```

Comment:
Suggest a comment here and in the commit description that informs reader of the two rules now enforced / implemented in this large block i.e. `null` / `bool` handling when there's just one value and `null` plus prefix handling in all other cases.